idseq_dag/steps/download_accessions.py

```diff
@@ -122,5 +149,21 @@ def fetch_sequence_for_thread(error_flags, accession, accession_out_file,
         finally:
             semaphore.release()
 
+    def generate_dummy_fasta_ref(self, output_reference_fasta):
+        with open(output_reference_fasta, 'w') as sf:
+            sf.write("test size")
+
+    def generate_size_file(self, accession_summary_tab, accession_dict, loc_db, accession_summary):
+        loc_dict = open_file_db_by_extension(loc_db, IdSeqDictValue.VALUE_TYPE_ARRAY)
 
+        with open(accession_summary_tab, 'w') as ast:
+            for accession_id, _ in accession_dict.items():
+                entry = loc_dict.get(accession_id)
+                if entry and len(entry) == 3:
+                    # The seq_len includes newline characters in the fasta file, so we must correct for this.
+                    seq_len = math.floor(FASTA_SEQ_LINE_LENGTH / (FASTA_SEQ_LINE_LENGTH + 1) * int(entry[2]))
```

Comment:
hm it's still not clear what this is.  also I think it's a bit dangerous to assume FASTA_SEQ_LINE_LENGTH is 80. Can you just make this a rough length?

Reply:
This calculates the number of nucleotide bases, and removes the newline characters from the count.

Good point about assuming 80. I will see if I can calculate dynamically in some way.

Reply:
If I understand correctly, the issue is that we cannot use the size of the byterange as the sequence length, because byterange includes new-line characters. I can see a few different options, some of which are much better than others:

(a) Calculate all the sequence lengths from the downloaded sequence files. This would be super wasteful to do at pipeline run time.
(b) Determine the actual sequence length while generating nt_loc_db (this is part of the index update operation) and store it in the dict once and for all so it can just be looked up. Probably the best option, but requires the development/testing overhead of building a new nt_loc_db.
(c) Keep the current setup with assumed line length of 80, but do a quick spot check in the code to confirm sequence files actually satisfy this condition. This code can probably live in the step that generates a new nt_loc_db when we update the index. Include an assertion with a very explicit / verbose message for the developers if the condition fails, so that they know how to address the problem (for example, "you need to change the value of FASTA_SEQ_LINE_LENGTH in download_accessions.py"). For what it's worth, I do think the line length of 80 will be a good assumption for a long time to come. Wikipedia article says it's something of a standard: https://en.wikipedia.org/wiki/FASTA_format

Anything I missed?